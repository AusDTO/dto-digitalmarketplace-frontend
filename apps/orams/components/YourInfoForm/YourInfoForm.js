/* eslint-disable */

import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Form } from 'react-redux-form'
import isNumber from 'lodash/isNumber'

import { required, validEmail, validPhoneNumber, minObjectLength } from 'shared/validators'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import Layout from 'shared/Layout'
import BaseForm from 'shared/form/BaseForm'
import SubmitForm from 'shared/form/SubmitForm'
import ErrorBox from 'shared/form/ErrorBox'
import Textfield from 'shared/form/Textfield'
import formProps from 'shared/form/formPropsSelector'
import { loadProfile } from 'orams/actions/profileActions'
import LoadingButton from 'shared/LoadingButton/LoadingButton'

class YourInfoForm extends BaseForm {
  static propTypes = {
    action: PropTypes.string,
    csrf_token: PropTypes.string,
    form: PropTypes.object.isRequired,
    supplierCode: PropTypes.number
  }

  componentDidMount() {
    this.props.loadProfileData(this.props.supplierCode, this.props.form.model)
  }

  render() {
    const {
      action,
      csrf_token,
      model,
      supplierCode,
      form,
      children,
      handleSubmit,
      nextRoute,
      submitClicked,
      profileUpdated,
      currentlySending
    } = this.props
    let title = 'Contact details'
    let hasFocused = false
    const setFocus = e => {
      if (!hasFocused) {
        hasFocused = true
        e.focus()
      }
    }
    return (
      <Layout>
        <header>
          <h1 className="au-display-xl" tabIndex="-1">
            {title}
          </h1>
        </header>
        <article role="main">
          {profileUpdated
            ? <AUpageAlert as="success">
                <h4>Profile updated</h4>
              </AUpageAlert>
            : ''}
          <ErrorBox model={model} setFocus={setFocus} submitClicked={submitClicked} />
          <Form model={model} action={action} id="yourinfo" validateOn="submit" onSubmit={data => handleSubmit(data)}>
            {csrf_token && <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />}

            <Textfield
              model={`${model}.contact_name`}
              name="contact_name"
              id="contact_name"
              htmlFor="contact_name"
              label="Business contact name"
              description="The contact listed on your seller profile page."
              validators={{ required }}
              messages={{
                required: 'Business contact is required'
              }}
            />

            <Textfield
              model={`${model}.contact_email`}
              name="contact_email"
              id="contact_email"
              htmlFor="contact_email"
              label="Email"
              validators={{ required, validEmail }}
              messages={{
                required: 'Business contact email is required',
                validEmail: 'Business contact email must be a valid email address'
              }}
            />

            <Textfield
              model={`${model}.contact_phone`}
              name="contact_phone"
              id="contact_phone"
              htmlFor="contact_phone"
              label="Phone"
              validators={{ required, validPhoneNumber }}
              messages={{
                required: 'Business contact phone is required',
                validPhoneNumber: 'Business contact phone number must be a valid phone number'
              }}
            />

            <Textfield
              model={`${model}.representative`}
              name="representative"
              id="representative"
              htmlFor="representative"
              label="Authorised representative name"
              description="This is the person authorised to enter into contracts on behalf of the business. "
              validators={{ required }}
              messages={{
                required: 'Authorised representative is required'
              }}
            />

            <Textfield
              model={`${model}.email`}
              name="email"
              id="email"
              htmlFor="email"
              label="Email"
              validators={{ required, validEmail }}
              messages={{
                required: "Authorised representative's email is required",
                validEmail: "Authorised representative's email must be a valid email address"
              }}
            />

            <Textfield
              model={`${model}.phone`}
              name="phone"
              id="phone"
              htmlFor="phone"
              label="Phone"
              validators={{ required, validPhoneNumber }}
              messages={{
                required: "Authorised representative's phone number is required",
                validPhoneNumber: "Authorised representative's phone number must be a valid phone number"
              }}
            />

            {children}

            {currentlySending
              ? <LoadingButton />
              : <button type="submit" className="au-btn">
                  Update profile
                </button>}
          </Form>
        </article>
      </Layout>
    )
  }
}

YourInfoForm.defaultProps = {
  title: 'Add your contact details'
}

const mapStateToProps = state => {
  return {
    ...formProps(state, 'yourInfoForm'),
    profileUpdated: state.app.profileUpdated,
    currentlySending: state.app.currentlySending,
    supplierCode: state.app.supplierCode
  }
}

const mapDispatchToProps = dispatch => {
  return {
    loadProfileData: (supplierCode, form) => dispatch(loadProfile(supplierCode, form))
  }
}

export { Textfield, mapStateToProps, YourInfoForm as Form }

export default connect(mapStateToProps, mapDispatchToProps)(YourInfoForm)
