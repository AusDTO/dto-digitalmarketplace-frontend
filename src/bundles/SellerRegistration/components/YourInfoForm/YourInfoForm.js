import React from 'react';
import { connect } from 'react-redux';
import { Form } from 'react-redux-form';

import { required, validEmail, validPhoneNumber } from '../../../../validators';

import Layout        from '../../../../shared/Layout';
import BaseForm      from '../../../../shared/form/BaseForm';
import SubmitForm    from '../../../../shared/form/SubmitForm';
import ErrorBox      from '../../../../shared/form/ErrorBox';
import Textfield     from '../../../../shared/form/Textfield';
import formProps     from '../../../../shared/reduxModules/formPropsSelector';

class YourInfoForm extends BaseForm {

  static propTypes = {
    action: React.PropTypes.string,
    csrf_token: React.PropTypes.string,
    form: React.PropTypes.object.isRequired,
    supplierCode: React.PropTypes.number
  }

  render() {
    const { action, csrf_token, model, supplierCode, userName, userEmail, form, buttonText, children, onSubmit, onSubmitFailed } = this.props;
    let title = 'Contact details';
    if (supplierCode) {
        title = 'Check your contact details'
    }

    return (
      <Layout>
        <header>
          <h1 tabIndex="-1">{title}</h1>
        </header>
        <article role="main">
          <ErrorBox focusOnMount={true} model={model}/>
          <Form model={model}
            action={action}
            method="post"
            id="yourinfo"
            component={SubmitForm}
            valid={form.valid}
            onCustomSubmit={onSubmit}
            onSubmitFailed={onSubmitFailed}
          >
            {csrf_token && (
              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
            )}

              <Textfield
                  model={`${model}.contact_name`}
                  name="contact_name"
                  id="contact_name"
                  htmlFor="contact_name"
                  label="Business contact"
                  description="The contact listed on your seller profile page and the person who receives new opportunities by email."
                  validators={{ required }}
                  messages={{
                      required: 'Business contact is required',
                  }}
                  default={userName}
              />

              <Textfield
                  model={`${model}.contact_email`}
                  name="contact_email"
                  id="contact_email"
                  htmlFor="contact_email"
                  label="Email"
                  validators={{ required, validEmail }}
                  messages={{
                      required: 'Business contact email is required',
                      validEmail: 'Business contact email must be a valid email address',
                  }}
                  default={userEmail}
              />


              <Textfield
                  model={`${model}.contact_phone`}
                  name="contact_phone"
                  id="contact_phone"
                  htmlFor="contact_phone"
                  label="Phone"
                  validators={{ required, validPhoneNumber }}
                  messages={{
                      required: 'Business contact phone is required',
                      validPhoneNumber: 'Business contact phone number must be a valid phone number',
                  }}
              />

              <Textfield
                  model={`${model}.representative`}
                  name="representative"
                  id="representative"
                  htmlFor="representative"
                  label="Authorised representative"
                  description="This is the person authorised to enter into contracts on behalf of the business. "
                  validators={{ required }}
                  messages={{
                      required: 'Authorised representative is required',
                  }}
              />


              <Textfield
                  model={`${model}.email`}
                  name="email"
                  id="email"
                  htmlFor="email"
                  label="Email"
                  validators={{ required, validEmail }}
                  messages={{
                      required: 'Authorised representative\'s email is required',
                      validEmail: 'Authorised representative\'s email must be a valid email address',
                  }}
              />

              <Textfield
                  model={`${model}.phone`}
                  name="phone"
                  id="phone"
                  htmlFor="phone"
                  label="Phone"
                  validators={{ required, validPhoneNumber }}
                  messages={{
                      required: 'Authorised representative\'s phone number is required',
                      validPhoneNumber: 'Authorised representative\'s phone number must be a valid phone number',
                  }}
              />

            {children}

            <input type="submit" value={buttonText} role="button" />
          </Form>
        </article>
      </Layout>
    )
  }
}

YourInfoForm.defaultProps = {
  buttonText: 'Save and continue',
  title: 'Add your contact details'
}

const mapStateToProps = (state) => {
  return {
      userName: state.form_options.user_name,
      userEmail: state.form_options.user_email,
      supplierCode: (state.application && state.application.supplierCode),
    ...formProps(state, 'yourInfoForm')
  }
}

export {
  Textfield,
  mapStateToProps,
  YourInfoForm as Form
}

export default connect(mapStateToProps)(YourInfoForm);
