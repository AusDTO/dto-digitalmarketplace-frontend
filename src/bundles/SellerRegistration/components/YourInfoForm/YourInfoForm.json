{
  "yourInfoForm": {
    "representative": "My Rep",
    "name": "The Company Name",
    "abn": "1234567",
    "phone": "5555555555",
    "email": "test@company.com"
  },
  "form_options": {
    "action": "https://httpbin.org/post"
  }
}
