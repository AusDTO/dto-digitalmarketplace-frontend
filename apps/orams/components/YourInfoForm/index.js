import YourInfoForm from './YourInfoForm'

export default YourInfoForm
