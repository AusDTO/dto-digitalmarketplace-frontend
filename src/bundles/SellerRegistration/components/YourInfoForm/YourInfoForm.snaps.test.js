// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');
//jest.mock('react-router');

import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router';
import renderer from 'react-test-renderer';

import YourInfoForm, { Textfield } from './YourInfoForm';
import createStore from '../../redux/create';

test('YourInfoForm renders', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <YourInfoForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('YourInfoForm renders with form_options', () => {
  const form_options = {
    csrf_token: 'sometoken',
    action: '/foo/bar'
  }
  let store = createStore(Object.assign({}, { _serverContext: {}, form_options }))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <YourInfoForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
