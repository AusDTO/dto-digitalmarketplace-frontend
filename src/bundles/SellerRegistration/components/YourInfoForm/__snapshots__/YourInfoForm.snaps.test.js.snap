// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourInfoForm renders 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      Contact details
    </h1>
  </header>
  <article
    role="main"
  >
    <form
      action={undefined}
      id="yourinfo"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_name"
        >
          Business contact name
        </label>
        <p
          className="hint"
          id="contact_name-hint"
        >
          The contact listed on your seller profile page and the person who receives new opportunities by email.
        </p>
        <input
          aria-describedby="contact_name-hint"
          className=""
          disabled={false}
          id="contact_name"
          maxLength={undefined}
          name="contact_name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_email"
        >
          Email
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="contact_email"
          maxLength={undefined}
          name="contact_email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_phone"
        >
          Phone
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="contact_phone"
          maxLength={undefined}
          name="contact_phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="representative"
        >
          Authorised representative name
        </label>
        <p
          className="hint"
          id="representative-hint"
        >
          This is the person authorised to enter into contracts on behalf of the business. 
        </p>
        <input
          aria-describedby="representative-hint"
          className=""
          disabled={false}
          id="representative"
          maxLength={undefined}
          name="representative"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="email"
        >
          Email
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="email"
          maxLength={undefined}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="phone"
        >
          Phone
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="phone"
          maxLength={undefined}
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Save and continue
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;

exports[`YourInfoForm renders with edit mode 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      Contact details
    </h1>
  </header>
  <article
    role="main"
  >
    <form
      action="/foo/bar"
      id="yourinfo"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <input
        id="csrf_token"
        name="csrf_token"
        type="hidden"
        value="sometoken"
      />
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_name"
        >
          Business contact name
        </label>
        <p
          className="hint"
          id="contact_name-hint"
        >
          The contact listed on your seller profile page and the person who receives new opportunities by email.
        </p>
        <input
          aria-describedby="contact_name-hint"
          className=""
          disabled={false}
          id="contact_name"
          maxLength={undefined}
          name="contact_name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_email"
        >
          Email
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="contact_email"
          maxLength={undefined}
          name="contact_email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_phone"
        >
          Phone
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="contact_phone"
          maxLength={undefined}
          name="contact_phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="representative"
        >
          Authorised representative name
        </label>
        <p
          className="hint"
          id="representative-hint"
        >
          This is the person authorised to enter into contracts on behalf of the business. 
        </p>
        <input
          aria-describedby="representative-hint"
          className=""
          disabled={false}
          id="representative"
          maxLength={undefined}
          name="representative"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="email"
        >
          Email
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="email"
          maxLength={undefined}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="phone"
        >
          Phone
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="phone"
          maxLength={undefined}
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Save and continue
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;

exports[`YourInfoForm renders with form_options 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      Contact details
    </h1>
  </header>
  <article
    role="main"
  >
    <form
      action="/foo/bar"
      id="yourinfo"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <input
        id="csrf_token"
        name="csrf_token"
        type="hidden"
        value="sometoken"
      />
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_name"
        >
          Business contact name
        </label>
        <p
          className="hint"
          id="contact_name-hint"
        >
          The contact listed on your seller profile page and the person who receives new opportunities by email.
        </p>
        <input
          aria-describedby="contact_name-hint"
          className=""
          disabled={false}
          id="contact_name"
          maxLength={undefined}
          name="contact_name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_email"
        >
          Email
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="contact_email"
          maxLength={undefined}
          name="contact_email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="contact_phone"
        >
          Phone
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="contact_phone"
          maxLength={undefined}
          name="contact_phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="representative"
        >
          Authorised representative name
        </label>
        <p
          className="hint"
          id="representative-hint"
        >
          This is the person authorised to enter into contracts on behalf of the business. 
        </p>
        <input
          aria-describedby="representative-hint"
          className=""
          disabled={false}
          id="representative"
          maxLength={undefined}
          name="representative"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="email"
        >
          Email
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="email"
          maxLength={undefined}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="field"
      >
        <label
          className="question-heading"
          htmlFor="phone"
        >
          Phone
        </label>
        <input
          aria-describedby={undefined}
          className=""
          disabled={false}
          id="phone"
          maxLength={undefined}
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          readOnly={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Save and continue
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;
